### PR TITLE
Use `@_implementationOnly import` for FirebaseCoreExtension

### PR DESCRIPTION
**Pull request by ryanwilson**
Using a regular import caused internal Core APIs to be surfaced to end
users.

### CODE REVIEW

**Review by paulb777 — APPROVED** [review]
Good find! Did you confirm the interop pods and FirebaseSharedSwift APIs are not exposed also?

File: FirebaseFunctions/Sources/Functions.swift [hunk 1]
```diff
@@ -25,6 +24,9 @@ import FirebaseSharedSwift
   import GTMSessionFetcherCore
 #endif
 
+// Avoids exposing internal FirebaseCore APIs to Swift users.
+@_implementationOnly import FirebaseCoreExtension
```
Comment (ncooke3) [inline FirebaseFunctions/Sources/Functions.swift:28]: Awesome!

**Review by ncooke3 — APPROVED** [review]

File: FirebaseFunctions/Sources/Functions.swift (hunk 1 above)
Comment (ryanwilson) [reply FirebaseFunctions/Sources/Functions.swift:28]: The leading underscore makes me nervous that it'll break at some point unexpectedly... but it's better than the alternative at least 🤷 .

**Comment by ryanwilson** [comment]
> Good find! Did you confirm the interop pods and FirebaseSharedSwift APIs are not exposed also?

Confirmed that interop pods are not exposed, neither is StorageInternal. 

Thanks for bringing up SharedSwift: the SharedSwift APIs are actually required to be exposed but currently aren't, so auto complete isn't working as expected. I'll see if we can address that with `@_exported` in a follow up PR.

**Pull request merged**